release/release_guide.md

```diff
@@ -0,0 +1,678 @@
+
+To prepare for each release, you should audit the project status in the JIRA issue tracker, and do the necessary
+bookkeeping. Finally, you should create a release branch from which individual release candidates will be built.
+
+**NOTE**: If you are
+using [GitHub two-factor authentication](https://help.github.com/articles/securing-your-account-with-two-factor-authentication-2fa/)
+and haven’t configure HTTPS access, please
+follow [the guide](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/) to configure
+command line access.
+
+## One-time Setup Instructions
+
+You need to have a GPG key to sign the release artifacts. Please be aware of the
+ASF-wide [release signing guidelines](https://www.apache.org/dev/release-signing.html). If you don’t have a GPG key
+associated with your Apache account, please follow the section below.
+
+### For Linux users
+
+There are 2 ways to configure your GPG key for release, either using release automation script(which is recommended), or
+running all commands manually. If using Mac, please see below to handle known issues.
+
+#### Use preparation_before_release.sh to setup GPG
+
+- Script: preparation_before_release.sh
+- Usage ./hudi/scripts/release/preparation_before_release.sh
+- Tasks included
+    1. Help you create a new GPG key if you want.
+    2. Configure git user.signingkey with chosen pubkey.
+    3. Add chosen pubkey into dev KEYS and release KEYS **NOTES**: Only PMC can write into release repo.
+    4. Start GPG agents.
+
+#### Run all commands manually
+
+- Get more entropy for creating a GPG key
+    - sudo apt-get install rng-tools
+    - sudo rngd -r /dev/urandom
+- Create a GPG key
+    - gpg --full-generate-key
+- Determine your Apache GPG Key and Key ID, as follows:
+    - gpg --list-keys
+- This will list your GPG keys. One of these should reflect your Apache account, for example:
+  ```
+  --------------------------------------------------
+  pub   2048R/935D191 2019-08-29
+  uid   Anonymous Anonymous <anonymous@apache.org>
+  sub   2048R/CD4C59FD 2019-08-29
+  ```
+  Here, the key ID is the 8-digit hex string in the pub line: 845E6689 or more than 8-digit hex string like
+  623E08E06DB376684FB9599A3F5953147903948A. Now, add your Apache GPG key to the Hudi’s KEYS file both
+  in [dev](https://dist.apache.org/repos/dist/dev/hudi/KEYS)
+  and [release](https://dist.apache.org/repos/dist/release/hudi/KEYS) repositories
+  at [dist.apache.org](http://dist.apache.org/). Follow the instructions listed at the top of these files. (**Note**:
+  Only PMC members have write access to the release repository. If you end up getting 403 errors ask on the mailing list
+  for assistance.)
+- Configure git to use this key when signing code by giving it your key ID, as follows:
+  - git config --global user.signingkey CD4C59FD, or git config --global user.signingkey
+    623E08E06DB376684FB9599A3F5953147903948A
```

Comment:
is this fake signing key or real one? lets mask if its real one.

Reply:
fixed.